core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -278,6 +281,46 @@ static ByteStreamMessage of(Path path, @Nullable ExecutorService executor, ByteB
         return builder.alloc(alloc).bufferSize(bufferSize).build();
     }
 
+    /**
+     * Creates a new {@link StreamMessage} that streams the specified {@link OutputStream}.
+     *
+     * <p>For example:<pre>{@code
+     * ByteStreamMessage byteStreamMessage = StreamMessage.fromOutputStream(os -> {
+     *     try {
+     *         for (int i = 0; i < 5; i++) {
+     *             os.write(i);
+     *         }
+     *         os.close();
+     *     } catch (IOException e) {
+     *         throw new RuntimeException(e);
+     *     }
+     * });
+     * byte[] result = byteStreamMessage.collectBytes().join();
+     *
+     * assert Arrays.equals(result, new byte[] { 0, 1, 2, 3, 4 });
+     * }</pre>
+     */
+    static ByteStreamMessage fromOutputStream(Consumer<OutputStream> outputStreamWriter) {
+        final RequestContext ctx = RequestContext.currentOrNull();
+        ExecutorService blockingTaskExecutor = null;
+        if (ctx instanceof ServiceRequestContext) {
+            blockingTaskExecutor = ((ServiceRequestContext) ctx).blockingTaskExecutor();
+        }
+        if (blockingTaskExecutor == null) {
+            blockingTaskExecutor = CommonPools.blockingTaskExecutor();
+        }
+        return fromOutputStream(outputStreamWriter, blockingTaskExecutor);
+    }
+
+    /**
+     * Creates a new {@link StreamMessage} that streams the specified {@link OutputStream}.
```

Comment:
ditto.
Could you also copy and past the example?

Reply:
✅ updated!